.github/workflows/haskell-release.yml

```diff
@@ -0,0 +1,115 @@
+name: Release
+
+on:
+  # Trigger the workflow on the new PVP version tag created.
+  # https://pvp.haskell.org/
+  # See patterns explanation at:
+  # https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions
+  push:
+    tags:
+      - "[0-9]+.[0-9]+.[0-9]+.[0-9]+*"
+
+jobs:
+  create_release:
+    name: Create Github Release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+
+      - name: Zip data directory
+        run: |
+          zip -r data.zip ./data || { echo "Unable to create a zip archive."; exit 1;  }
+
+      - name: Create Release
+        id: create_release
+        uses: softprops/action-gh-release@v1
+        with:
+          files: |
+            LICENSE
+            data.zip
+
+      - name: Output Release URL File
+        run: echo "${{ steps.create_release.outputs.upload_url }}" > release_url.txt
+
+      - name: Save Release URL File for publish
+        uses: actions/upload-artifact@v3
+        with:
+          name: release_url
+          path: release_url.txt
+
+  build_artifact:
+    needs: [create_release]
+    name: Haskell-Release - ${{ matrix.os }} - ${{ matrix.compiler }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        # os: [ubuntu-latest, macOS-latest, windows-latest]
+        # ghc:
+        #   - "8.8.3"
+        #   - "8.10.1"
+        # cabal: ["3.2"]
+        include:
+          - compiler: ghc-8.10.7
+            compilerKind: ghc
+            compilerVersion: 8.10.7
+            cabal: "3.6"
+            # It might be good idea to use older runner for building binary:
+            # on latest runners like fedora, it would link symbols in newer
+            # system libraries, so users could not run it on older OSes
+            os: ubuntu-20.04
```

Comment:
Can we somehow use this to build on other OSes besides just Ubuntu?

Reply:
Like with another item in the `include` section? Hopefully yes, I would like to add macOS with GHC 9.2 in the future.

Reply:
Right, I meant like macOS and Windows, I realize the Ubuntu build will probably work on other GNU/Linux variants.

Reply:
Well, I guess Windows doesn't matter since you would need to run Swarm under something like WSL anyway.

Reply:
Yes, but macOS might be worth a little effort.

At the moment [GitHub does not have ARM macOS runners](https://github.com/actions/runner-images/issues/2187), so maybe we can wait a little while longer.